### ISSUE TITLE
componentWillReceiveProps warning in React 16.9.0

### ISSUE THREAD

**Issue by Hurtak** [airbnb/react-outside-click-handler#30]
Since React 16.9.0 this package will produce following warning into the console

```
Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: OutsideClickHandler
```

**Comment by ljharb** [comment]
Since this package supports many more versions of reach than just 16.3+, I’m not sure how that warning is avoidable.

**Comment by Hurtak** [comment]
By briefly looking at the code, changing `componentWillReceiveProps` to `componentDidUpdate` could work without any behavior change. Did not do a deep dive so I could be wrong.

**Comment by ljharb** [comment]
cWRP runs before the update, cDU runs after, so I'm not sure that will be sufficient.

**Comment by Hurtak** [comment]
I know the difference betwenn the methods, my point was that by looking about code inside the cWRP, the behavior of the component will probably not change (I did not test this) and should be safe to move to cDU

**Comment by ljharb** [comment]
If it can be programmatically tested, or if you can make a persuasive argument that you've tested it and that that is true, then I will be happy to make the change.

**Comment by skt-t1-byungi** [comment]
"componentDidUpdate" is sufficient. "componentDidUpdate" did the actual DOM update, but since it is called before idle, the dom is not drawn to the screen.

**Comment by ljharb** [comment]
Fixed in #36.
